truckersmp_cli/main.py

```diff
@@ -186,21 +209,49 @@ def start_with_proton():
 
     prefix = os.path.join(Args.prefixdir, "pfx")
     proton = os.path.join(Args.protondir, "proton")
-    argv = [sys.executable, proton, "run"]
+    major, minor = get_proton_version(Args.protondir)
+    logging.info("Proton version is (major=%d, minor=%d)", major, minor)
+    proton_args = []
+    run_in_steamrt = []
+    if not Args.without_steam_runtime and (major >= 6 or (major == 5 and minor >= 13)):
+        # use Steam Runtime container for Proton 5.13+
+        logging.info("Using Steam Runtime container")
+        # share directories with Steam Runtime container
+        shared_paths = [Args.gamedir, Args.protondir, Args.prefixdir]
+        if not Args.singleplayer:
+            shared_paths += [Args.moddir, Dir.truckersmp_cli_data, Dir.scriptdir]
+        discord_sockets = find_discord_ipc_sockets()
+        if len(discord_sockets) > 0:
+            shared_paths += discord_sockets
+        logging.debug("Shared paths: %s", shared_paths)
+        run_in_steamrt.append(os.path.join(Args.steamruntimedir, "run"))
+        for shared_path in shared_paths:
+            run_in_steamrt += ["--filesystem", shared_path]
+        run_in_steamrt += ["--", "python3"]  # helper script
+        proton_args += ["--", "python3"]     # Proton script
+    else:
+        # don't use Steam Runtime container for older Proton
+        logging.info("Not using Steam Runtime container")
+        run_in_steamrt.append(sys.executable)  # helper
+        proton_args += ["--", sys.executable]  # Proton
+    wine = run_in_steamrt.copy()
+    proton_args += [proton, "run"]
+
     env = os.environ.copy()
     env["STEAM_COMPAT_DATA_PATH"] = Args.prefixdir
     env["STEAM_COMPAT_CLIENT_INSTALL_PATH"] = steamdir
 
     # Proton's "dist" directory tree is missing until first run
     # make sure it's present for using "dist/bin/wine" directly
-    wine = os.path.join(Args.protondir, "dist/bin/wine")
-    if (not os.access(wine, os.R_OK)
+    wine_command = os.path.join(Args.protondir, "dist/bin/wine")
+    wine.append(wine_command)
+    if (not os.access(wine_command, os.R_OK)
             # native d3dcompiler_47 is removed when the prefix is downgraded
             # make sure the prefix is already upgraded/downgraded
             or Args.activate_native_d3dcompiler_47):
         try:
             subproc.check_output(
-                argv + ["wineboot", ], env=env, stderr=subproc.STDOUT)
+                proton_args + ["wineboot", ], env=env, stderr=subproc.STDOUT)
```

Comment:
With a fresh downloaded proton:

~~~
[…]
** INFO **  Using Steam Runtime container
** DEBUG **  Shared paths: ['/mnt/gamehub/native2proton/Euro Truck Simulator 2/', '/home/lucki/.local/share/truckersmp-cli/Proton', '/home/lucki/.local/share/truckersmp-cli/Euro Truck Simulator 2/prefix']
Failed to run wineboot: [Errno 2] File or directory not found: '--'
~~~

`proton_args`:
~~~
['--', 'python3', '/home/lucki/.local/share/truckersmp-cli/Proton/proton', 'run'] + ["wineboot", ]
~~~